### PR TITLE
fix(#1777): filter statuses for transfers

### PR DESCRIPTION
**Pull request by vomikan**
This change is [Reviewable](https://reviewable.io/reviews/moneymanagerex/moneymanagerex/1837)

### CODE REVIEW

**Comment by slodki** [comment]
What do you thing about this:

```c++
if (account_selected_in_filter) { // transaction as viewed from selected account
  switch (filterStatus) {
    case "N": return itemStatus.empty() || itemStatus.StartsWith("N");
    case "U": return itemStatus.empty() || itemStatus.StartsWith("N") || itemStatus.StartsWith("F");
    case "A": return !itemStatus.StartsWith("R");
    default: return itemStatus.StartsWith(filterStatus);
  }
} else { // all accounts
  switch (filterStatus) {
    case "N": return itemStatus.empty() || itemStatus.Contains("N");
    case "U": return itemStatus.empty() || itemStatus.Contains("N") || itemStatus.Contains("F");
    case "A": return !itemStatus.Contains("R");
    default: return itemStatus.Contains(filterStatus);
  }
}
```
And description for filters U and A should be changed to something easy to understand.

**Comment by vomikan** [comment]
@slodki it does not working

image

**Comment by slodki** [comment]
Above example is not C++ code - this is an idea only. And still not good.
`if (account_selected_in_filter) ` is still not enough - this code must distinct between *from* account and *to* account...
I will describe the new approach here...

**Comment by vomikan** [comment]
@slodki I suggest to commit this code as is. It's a bug fix.
Then you can reorginize it as you wish.

File: src/filtertransdialog.cpp [hunk 1]
```diff
@@ -86,6 +86,7 @@ mmFilterTransactionsDialog::mmFilterTransactionsDialog(wxWindow* parent, int acc
     , payeeID_(-1)
     , refAccountID_(account_id)
     , refAccountStr_("")
+    , m_filterStatus("")
```
Comment (slodki) [inline src/filtertransdialog.cpp]: ‘mmFilterTransactionsDialog::m_filterStatus’ will be initialized after ‘double mmFilterTransactionsDialog::m_min_amount’ when initialized here [-Wreorder]

**Pull request merged**